Branch-SDK-Tests/BNCFacebookAppLinksTests.m

```diff
@@ -0,0 +1,59 @@
+//
+//  BNCFacebookAppLinksTests.m
+//  Branch-SDK-Tests
+//
+//  Created by Ernest Cho on 10/24/19.
+//  Copyright © 2019 Branch, Inc. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "BNCFacebookMock.h"
+#import "BNCFacebookAppLinks.h"
+
+@interface BNCFacebookAppLinks()
+- (BOOL)isDeepLinkingClassAvailable;
+@end
+
+@interface BNCFacebookAppLinksTests : XCTestCase
+@property (nonatomic, strong, readwrite) BNCFacebookAppLinks *applinks;
+@end
+
+@implementation BNCFacebookAppLinksTests
+
+- (void)setUp {
+    self.applinks = [BNCFacebookAppLinks new];
+}
+
+- (void)tearDown {
+
+}
+
+- (void)testIsDeepLinkingClassAvailable {
+    XCTAssertFalse([self.applinks isDeepLinkingClassAvailable]);
+}
+
+- (void)testRegisterFacebookDeepLinkingClass_String {
+    [self.applinks registerFacebookDeepLinkingClass:@"HelloWorld"];
+    XCTAssertFalse([self.applinks isDeepLinkingClassAvailable]);
+}
+
+- (void)testRegisterFacebookDeepLinkingClass_Mock {
+    [self.applinks registerFacebookDeepLinkingClass:[BNCFacebookMock new]];
+    XCTAssertTrue([self.applinks isDeepLinkingClassAvailable]);
+}
+
+- (void)testFetchFacebookAppLink {
+    __block XCTestExpectation *expectation = [self expectationWithDescription:@""];
+    
+    [self.applinks registerFacebookDeepLinkingClass:[BNCFacebookMock new]];
+    [self.applinks fetchFacebookAppLinkWithCompletion:^(NSURL * _Nullable appLink, NSError * _Nullable error) {
+        XCTAssertTrue([[appLink absoluteString] isEqualToString:@"https://branch.io"]);
+        [expectation fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:2 handler:^(NSError * _Nullable error) {
```

Comment:
completion block is optional, right? 

> An optional XCWaitCompletionHandler block to invoke when all expectations have

https://developer.apple.com/documentation/xctest/xctestcase/1500748-waitforexpectationswithtimeout?language=objc

Reply:
It is, I usually use an empty block instead of nil.

Reply:
Probably cause old ObjC code didn't have nullability info and calling a nil block crashes.  Basically a legacy coding habit that's not really necessary anymore.